sql-aggregates/profit-margin.sql

```diff
@@ -0,0 +1,33 @@
+with "revenue" as (
+    select "films"."filmId",
+          "films"."title",
+          sum("payments"."amount") as "totalRevenue"
+    from "inventory"
+    join "films" using ("filmId")
+    join "rentals" using ("inventoryId")
+    join "payments" using ("rentalId")
+    group by "films"."filmId"
+    order by "films"."filmId"
+),
+
+"cost" as (
+    select "films"."filmId",
+          "films"."title",
+          "films"."replacementCost",
+          count("inventory"."filmId") as "filmCount",
+          "films"."replacementCost" * count("inventory"."filmId") as "totalCost"
+    from "inventory"
+    join "films" using ("filmId")
+    group by "films"."filmId"
+    order by "films"."filmId"
+)
```

Comment:
The sub-queries performed by the common table expressions don't need their own sorts:
- `"filmId"` is the primary key for the `"films"` table, and tables are initially sorted by their primary key, so these `order by`s are redundant.
- The database engine comes up with its own query strategy which matches up corresponding rows from different tables, so forcing it to sort temporary table performs unnecessary work.